Packs/ApiModules/Scripts/CoreIRApiModule/CoreIRApiModule.py

```diff
@@ -3339,3 +3269,192 @@ def get_dynamic_analysis_command(client: CoreClient, args: Dict) -> CommandResul
+
+    if last_seen_gte:
+        filters.append({
+            'field': 'last_seen',
+            'operator': 'gte',
+            'value': last_seen_gte
+        })
+
+    if last_seen_lte:
+        filters.append({
+            'field': 'last_seen',
+            'operator': 'lte',
+            'value': last_seen_lte
+        })
+
+    return filters
+
+
+def args_to_request_filters(args):
+
+    if set(args.keys()) & {  # check if any filter argument was provided
+        'endpoint_id_list', 'dist_name', 'ip_list', 'group_name', 'platform', 'alias_name',
+        'isolate', 'hostname', 'status', 'first_seen_gte', 'first_seen_lte', 'last_seen_gte', 'last_seen_lte'
+    }:
+        endpoint_id_list = argToList(args.get('endpoint_id_list'))
+        dist_name = argToList(args.get('dist_name'))
+        ip_list = argToList(args.get('ip_list'))
+        group_name = argToList(args.get('group_name'))
+        platform = argToList(args.get('platform'))
+        alias_name = argToList(args.get('alias_name'))
+        isolate = args.get('isolate')
+        hostname = argToList(args.get('hostname'))
+        status = args.get('status')
+
+        first_seen_gte = arg_to_timestamp(
+            arg=args.get('first_seen_gte'),
+            arg_name='first_seen_gte'
+        )
+
+        first_seen_lte = arg_to_timestamp(
+            arg=args.get('first_seen_lte'),
+            arg_name='first_seen_lte'
+        )
+
+        last_seen_gte = arg_to_timestamp(
+            arg=args.get('last_seen_gte'),
+            arg_name='last_seen_gte'
+        )
+
+        last_seen_lte = arg_to_timestamp(
+            arg=args.get('last_seen_lte'),
+            arg_name='last_seen_lte'
+        )
+
+        return create_request_filters(
+            endpoint_id_list=endpoint_id_list, dist_name=dist_name, ip_list=ip_list,
+            group_name=group_name, platform=platform, alias_name=alias_name, isolate=isolate, hostname=hostname,
+            first_seen_lte=first_seen_lte, first_seen_gte=first_seen_gte,
+            last_seen_lte=last_seen_lte, last_seen_gte=last_seen_gte, status=status
+        )
+    # a request must be sent with at least one filter parameter, so by default we will send the endpoint_id_list filter
+    return create_request_filters(endpoint_id_list=argToList(args.get('endpoint_ids')))
+
+
+def add_tag_to_endpoints_command(client: CoreClient, args: Dict):
+    endpoint_ids = argToList(args.get('endpoint_ids', []))
+    tag = args.get('tag')
+
+    filters = args_to_request_filters(args)
+    raw_response = client.add_or_remove_tag_endpoint(endpoint_ids=endpoint_ids, tag=tag, filters=filters)
+
+    return CommandResults(
+        readable_output=f'Successfully added tag {tag} to endpoint(s) {endpoint_ids}', raw_response=raw_response
+    )
+
+
+def remove_tag_from_endpoints_command(client: CoreClient, args: Dict):
+    endpoint_ids = argToList(args.get('endpoint_ids', []))
+    tag = args.get('tag')
+
+    filters = args_to_request_filters(args)
```

Comment:
args_to_request_filters could probably be called in add_or_remove_tag_endpoint since its common. Matter of choice

Reply:
will call it inside `call_tag_endpoint` method